src/components/pages/Contacts.jsx

```diff
@@ -1,10 +1,17 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import HeaderWithNav from './header';
 
-export default function Contacts() {
+export default function Contacts(props) {
+  const { isAd } = props;
+
```

Comment:
Currently, a non admin can get to this (and the history) page by accessing the URL. To fix this, just use something like this:

  if (isAdmin) {
    return (
      <div>
        <HeaderWithNav isAdmin={isAdmin}>Contacts</HeaderWithNav>
      </div>
    );
  }

  return (
    <div> You do not have access to this page </div>
  );

Reply:
You can also change the ProtectedRoutes in App.js to check if the user is an admin by changing 
`<PrivateRoute isLoggedIn={isUserLoggedIn()} path="/history"><History /></PrivateRoute>` to 
`<PrivateRoute isLoggedIn={isAdmin} path="/history"><History /></PrivateRoute>` 
in order to achieve the same effect. If you've already implemented Cole's solution that works too though.

Reply:
Yeah, the way Vance mentioned above is a better way to do it than what I previously said.